geppetto.js/geppetto-ui/src/3d-canvas/showcase/examples/SelectionUtils.js

```diff
@@ -0,0 +1,40 @@
+import { hexToRGBNormalized } from "../../threeDEngine/util";
+
+const SELECTION_COLOR = { r: 0.8, g: 0.8, b: 0, a: 1 };
+
+
+export function dataMapping (data){
+  return data.map(item => (
+    {
+      color: item.selected ? SELECTION_COLOR : item.color ? item.color : hexToRGBNormalized(GEPPETTO.Resources.COLORS.DEFAULT),
+      instancePath: item.instancePath
+    }
+  ))
+}
+
+export function onSelection (selectedInstances) {
```

Comment:
good idea to generalize it for the showcase

Better avoid to use this.state or setState in utility methods outside of the component class. 
Rather limit this function's knowledge to data, selectedInstances and newData. 

E.g.: `applySelection(data, instances)` -> `newData`

Even if that means that each showcase Component will define a short function like this:
```
const onSelection = (selectedInstances) =>  this.setState({data: Utility.applySelection(this.data, selectedInstances)}`
```